newIDE/app/src/ObjectsRendering/PixiResourcesLoader.js

```diff
@@ -232,6 +234,72 @@ export default class PixiResourcesLoader {
     return material;
   }
 
+  /**
+   * Return the three.js material associated to the specified resource name.
+   * @param project The project
+   * @param resourceName The name of the resource
+   * @param options
+   * @returns The requested material.
+   */
+  static get3DModel(
+    project: gdProject,
+    resourceName: string
+  ): Promise<THREE.Object3D> {
+    const loaded3DModel = loaded3DModels[resourceName];
+    if (loaded3DModel) return Promise.resolve(loaded3DModel);
+
+    if (!project.getResourcesManager().hasResource(resourceName))
+      return invalidTexture;
+
+    const resource = project.getResourcesManager().getResource(resourceName);
+    if (resource.getKind() !== 'model3D') return invalidTexture;
+
+    const url = ResourcesLoader.getResourceFullUrl(project, resourceName, {
+      isResourceForPixi: false,
+    });
+
+    const loader = new GLTFLoader();
+    // TODO Cache promises that are not yet resolved to void `load` being
+    // called more than once for the same resource.
```

Comment:
Yes, we need a `PromiseCache` or something like this. I thought we had a helper for it. Something were you instantiate a cache as an object, then you can call it with a key (the url) and the function to call (`() =>loader.load(...)` if there is a cache miss. And the result promise is stored once, and when it resolves, it calls all the promises stored). If there is a cache hit, a new promise is returned (which will be resolved when the function resolves).

Reply:
I think I remember doing something like this on the Runtime side maybe for JSON.

Reply:
Indeed there is something a bit similar (though less elegant because based on callbacks - we could refactor this to return a promise if no callback is set).
In case of a promise, I think it's a bit simpler because you just store the promise and then return it, and then the caller can do a `then` (or await) on it.